### PR TITLE
Remove  the otlp trace exporter limit of SpanEvents when exporting

### LINKED ISSUE #2607: OTLP exporter limited to 128 SpanEvents
### Description

The OTLP exporter has a hardcoded const limit of 128 SpanEvents.

The TraceProvider can be configured with custom SpanLimits, which includes a higher EventLimit: https://github.com/open-telemetry/opentelemetry-go/blob/98bb1056c028ff816ed33b12fad7ab8661f8f9c7/sdk/trace/tracer.go#L138

However if you use the OTLP exporter to export the traces, your Events will be truncated to 128: 
https://github.com/open-telemetry/opentelemetry-go/blob/98bb1056c028ff816ed33b12fad7ab8661f8f9c7/exporters/otlp/otlptrace/internal/tracetransform/span.go#L27
https://github.com/open-telemetry/opentelemetry-go/blob/98bb1056c028ff816ed33b12fad7ab8661f8f9c7/exporters/otlp/otlptrace/internal/tracetransform/span.go#L180-L182

Given the fact that the EventLimit is enforced when capturing, I don't think any limit should be enforced when exporting.

### Environment

- OS: N/A
- Architecture: N/A
- Go Version: N/A
- opentelemetry-go version: main

### Steps To Reproduce

1. Set `EventLimit` to any value > 128
2. Export to somewhere using the OTLP exporter
3. Notice that only the first 128 events are present on a span

### Expected behavior

All events up to the configured limit should be exported.

### PR DESCRIPTION
**Pull request by hanyuancheung**
Fix #2607 

Remove the limit of SpanEvents when exporting.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,18 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 ## [Unreleased]
 
+### Added
+
+### Changed
+
+### Deprecated
+
+### Fixed
+
```
Comment (XSAM) [inline CHANGELOG.md:18]:
```suggestion
```
nil

File: exporters/otlp/otlptrace/internal/tracetransform/span.go [hunk 2]
```diff
@@ -177,17 +173,11 @@ func spanEvents(es []tracesdk.Event) []*tracepb.Span_Event {
 	}
 
 	evCount := len(es)
-	if evCount > maxEventsPerSpan {
-		evCount = maxEventsPerSpan
-	}
 	events := make([]*tracepb.Span_Event, 0, evCount)
```
Comment (XSAM) [inline exporters/otlp/otlptrace/internal/tracetransform/span.go:176]:
```suggestion
	events := make([]*tracepb.Span_Event, 0, len(es))
```

File: exporters/otlp/otlptrace/internal/tracetransform/span.go [hunk 3]
```diff
@@ -177,17 +173,11 @@ func spanEvents(es []tracesdk.Event) []*tracepb.Span_Event {
...
 	nEvents := 0
```
Comment (XSAM) [inline exporters/otlp/otlptrace/internal/tracetransform/span.go:177]:
```suggestion
```

File: exporters/otlp/otlptrace/internal/tracetransform/span.go [hunk 4]
```diff
@@ -177,17 +173,11 @@ func spanEvents(es []tracesdk.Event) []*tracepb.Span_Event {
...
 
 	// Transform message events
 	for _, e := range es {
-		if nEvents >= maxEventsPerSpan {
-			break
-		}
 		nEvents++
```
Comment (XSAM) [inline exporters/otlp/otlptrace/internal/tracetransform/span.go:181]:
```suggestion
```

**Review by XSAM — APPROVED** [review]

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Removed
+
+- Remove the limit of SpanEvents when exporting. (#2616)
```
Comment (pellared) [inline CHANGELOG.md:13]: Not sure if it should not be in `Fixed` section instead

Comment (XSAM) [reply CHANGELOG.md:13]: Yeah, I think putting this in the `Fixed` section is better.

Comment (hanyuancheung) [reply CHANGELOG.md]: done. Already changed. PTAL

**Review by MadVikingGod — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Fixed
+
+- Remove the limit of SpanEvents when exporting. (#2616)
```
Comment (MrAlias) [inline CHANGELOG.md:13]:
```suggestion
- Remove the OTLP trace exporter limit of SpanEvents when exporting. (#2616)
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**